From 6c075b6d265ed4dbac156d03406acc92fe066b93 Mon Sep 17 00:00:00 2001
From: Jason Hrycay <jason.hrycay@motorola.com>
Date: Fri, 26 Jul 2013 08:09:15 -0500
Subject: [PATCH 017/959] IKKRNBSP-2122 arm: msm: build: Add option to build
 motorola dtbs

Introduce a new configuration MMI_DEVICE_DTBS that will allow the
target to only build Motorola devices. If this option is configured
we will omit the QC architecture defaults for msm8916. This also will
enable adding 1kb of padding to the DTB to allow for insertion of
variables into the /chosen node at runtime in the bootloader.

Change-Id: I038a6d77bd4b827c9881f59c9c61ac8c5c62d766
[jasonh: Adjusted for msm8916]
Signed-off-by: Jason Hrycay <jason.hrycay@motorola.com>
Signed-off-by: Jeffrey Carlyle <jeff.carlyle@motorola.com>
---
 arch/arm/boot/dts/qcom/Makefile | 10 ++++++++++
 drivers/soc/qcom/Kconfig        |  8 ++++++++
 2 files changed, 18 insertions(+)

diff --git a/arch/arm/boot/dts/qcom/Makefile b/arch/arm/boot/dts/qcom/Makefile
index 73544303d09..c8c6b8dc6b4 100644
--- a/arch/arm/boot/dts/qcom/Makefile
+++ b/arch/arm/boot/dts/qcom/Makefile
@@ -65,6 +65,8 @@ dtb-$(CONFIG_ARCH_MSM8909) += msm8909-sim.dtb \
 	msm8208-1gb-rcm.dtb \
 	msm8208-qhd-cdp.dtb \
 	msm8208-qhd-rcm.dtb
+# Do not include QC default DTBs if not needed
+ifneq ($(CONFIG_MMI_DEVICE_DTBS),y)
 dtb-$(CONFIG_ARCH_MSM8916) += msm8916-sim.dtb \
 	msm8916-rumi.dtb \
 	msm8916-cdp.dtb \
@@ -164,6 +166,8 @@ dtb-$(CONFIG_ARCH_MSM8916) += msm8916-sim.dtb \
 	msm8929-qrd-skul.dtb \
 	msm8929-qrd-skulb.dtb \
 	msm8929-cdp-jdi.dtb
+endif # CONFIG_MMI_DEVICE_DTBS
+
 dtb-$(CONFIG_ARCH_MSM8226) += msm8226-sim.dtb \
 	msm8226-fluid.dtb \
 	msm8226-v1-cdp.dtb \
@@ -316,6 +320,12 @@ dtb-$(CONFIG_ARCH_MSM8992) += msm8992-sim.dtb \
 	msm8992-pmi8994-pm8004-mtp.dtb
 dtb-$(CONFIG_ARCH_MDMFERMIUM) += mdmfermium-rumi.dtb
 
+ifeq ($(CONFIG_MMI_DEVICE_DTBS),y)
+# Add 1k of padding to the DTBs to allow for environment variables
+# to be runtime added by the bootloader (i.e. /chosen properties)
+DTC_FLAGS := -p 1024
+endif
+
 DTB_NAMES := $(subst $\",,$(CONFIG_BUILD_ARM_APPENDED_DTB_IMAGE_NAMES))
 ifneq ($(DTB_NAMES),)
 DTB_LIST := $(addsuffix .dtb,$(DTB_NAMES))
diff --git a/drivers/soc/qcom/Kconfig b/drivers/soc/qcom/Kconfig
index caa109f2394..adc3af6131b 100644
--- a/drivers/soc/qcom/Kconfig
+++ b/drivers/soc/qcom/Kconfig
@@ -707,4 +707,12 @@ config MSM_VSENS
 
 source "drivers/soc/qcom/memshare/Kconfig"
 
+config MMI_DEVICE_DTBS
+ 	depends on ARCH_MSM
+ 	default n
+ 	bool "Motorola Mobility Device DTBs"
+ 	help
+ 	  Only include Motorola device specific DTBs to limit the number that
+	  get built during kernel compilation.
+
 endif # ARCH_MSM
-- 
2.11.0

